core/src/main/java/hivemall/evaluation/FMeasureUDAF.java

```diff
@@ -18,118 +18,387 @@
+        @Override
+        public Object terminatePartial(@SuppressWarnings("deprecation") AggregationBuffer agg)
+                throws HiveException {
+            FMeasureAggregationBuffer myAggr = (FMeasureAggregationBuffer) agg;
+
+            Object[] partialResult = new Object[5];
+            partialResult[0] = new LongWritable(myAggr.tp);
+            partialResult[1] = new LongWritable(myAggr.totalActual);
+            partialResult[2] = new LongWritable(myAggr.totalPredicted);
+            partialResult[3] = new DoubleWritable(myAggr.beta);
+            partialResult[4] = myAggr.average;
+            return partialResult;
         }
 
-        public DoubleWritable terminate() {
+        @Override
+        public void merge(@SuppressWarnings("deprecation") AggregationBuffer agg, Object partial)
+                throws HiveException {
             if (partial == null) {
-                return null;
+                return;
             }
-            double score = f1Score(partial);
-            return WritableUtils.val(score);
+
+            Object tpObj = internalMergeOI.getStructFieldData(partial, tpField);
+            Object totalActualObj = internalMergeOI.getStructFieldData(partial, totalActualField);
+            Object totalPredictedObj = internalMergeOI.getStructFieldData(partial,
+                totalPredictedField);
+            Object betaObj = internalMergeOI.getStructFieldData(partial, betaOptionField);
+            Object averageObj = internalMergeOI.getStructFieldData(partial, averageOptionFiled);
+            long tp = PrimitiveObjectInspectorFactory.writableLongObjectInspector.get(tpObj);
+            long totalActual = PrimitiveObjectInspectorFactory.writableLongObjectInspector.get(totalActualObj);
+            long totalPredicted = PrimitiveObjectInspectorFactory.writableLongObjectInspector.get(totalPredictedObj);
+            double beta = PrimitiveObjectInspectorFactory.writableDoubleObjectInspector.get(betaObj);
+            String average = PrimitiveObjectInspectorFactory.writableStringObjectInspector.getPrimitiveJavaObject(averageObj);
+
+            FMeasureAggregationBuffer myAggr = (FMeasureAggregationBuffer) agg;
+            myAggr.merge(tp, totalActual, totalPredicted, beta, average);
         }
 
-        /**
-         * @return 2 * precision * recall / (precision + recall)
-         */
-        private static double f1Score(final PartialResult partial) {
-            double precision = precision(partial);
-            double recall = recall(partial);
-            double divisor = precision + recall;
+        @Override
+        public DoubleWritable terminate(@SuppressWarnings("deprecation") AggregationBuffer agg)
+                throws HiveException {
+            FMeasureAggregationBuffer myAggr = (FMeasureAggregationBuffer) agg;
+            double result = myAggr.get();
+            return new DoubleWritable(result);
+        }
+    }
+
+    public static class FMeasureAggregationBuffer extends
```

Comment:
`estimate` is required for estimating resulting size for `AbstractAggregationBuffer`.

Refer
https://github.com/apache/hive/blob/master/ql/src/java/org/apache/hadoop/hive/ql/udf/generic/GenericUDAFComputeStats.java#L195